src/main/java/com/pimp/domain/Calendar.java

```diff
@@ -78,4 +79,16 @@ public boolean isPrivate() {
   public void setPrivate(boolean aPrivate) {
     isPrivate = aPrivate;
   }
+
+  public Event getEventByKey(String eventKey) {
+    Optional<Event> firstEvent = getEvents()
+      .stream()
+      .filter(event -> event.getKey().equals(eventKey))
+      .findFirst();
+    return firstEvent.isPresent() ? firstEvent.get() : null;
+  }
+
+  public boolean containsEvent(String eventKey) {
```

Comment:
Since this was a checking between an original event and a changed event, this wouldn't achieve the desired goal. nevertheless this might be superfluous with the change below.